bin/reth/src/args/utils.rs

```diff
@@ -44,8 +44,11 @@ pub fn genesis_value_parser(s: &str) -> eyre::Result<Arc<ChainSpec>, eyre::Error
         "holesky" => HOLESKY.clone(),
         "dev" => DEV.clone(),
         _ => {
-            let raw = fs::read_to_string(PathBuf::from(shellexpand::full(s)?.into_owned()))?;
-            let genesis: AllGenesisFormats = serde_json::from_str(&raw)?;
+            let genesis: AllGenesisFormats =
+                match fs::read_to_string(PathBuf::from(shellexpand::full(s)?.into_owned())) {
+                    Ok(raw) => serde_json::from_str(&raw)?, // from path
+                    Err(_) => serde_json::from_str(&s)?,    // from memory
```

Comment:
this will now result in a serde error if a path to a file is provided that does not exist

I agree we should always try to read the file first (though we could check if `s` starts with `{` to determine the order here)

but I'd assume providing raw json here is less likely than a file path that is wrong.

so we should return the io error here

Reply:
Good point. I changed the match statement to check the path first and then check that the string contains "{". Some of the examples from other tests use the `r# \n {...}` approach, which causes the valid json's first character to be `\n`, so I'm just using `.contains()` instead of `.starts_with()`